clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -2515,29 +2571,46 @@ scheduler_tmpl_t<RecordType, ReaderType>::pick_next_input(output_ordinal_t outpu
                                    output, target->index);
                             --num_blocked_;
                             target->blocked_time = 0;
+                            target->unscheduled = false;
                         }
+                    } else if (unscheduled_priority_.find(target)) {
+                        target->unscheduled = false;
+                        unscheduled_priority_.erase(target);
```

Comment:
No need to do anything.  The q counter is only used when an input is in a q which this is not; if it's added to a q it will get a new counter at that point.

Reply:
Added a note to input_info_t.queue_counter_.